homeassistant/components/plex/config_flow.py

```diff
@@ -225,6 +230,43 @@ def __init__(self):
         _LOGGER.debug("Imported Plex configuration")
         return await self.async_step_server_validate(import_config)
 
+    async def async_step_plex_website_auth(self):
+        """Begin external auth flow on Plex website."""
+        payload = {
+            "X-Plex-Device-Name": X_PLEX_DEVICE_NAME,
+            "X-Plex-Version": X_PLEX_VERSION,
+            "X-Plex-Product": X_PLEX_PRODUCT,
+            "X-Plex-Device": self.hass.config.location_name,
+            "X-Plex-Platform": X_PLEX_PLATFORM,
+            "X-Plex-Model": "Plex OAuth",
+        }
+        self.plexauth = PlexAuth(payload)
+        await self.plexauth.initiate_auth()
+        auth_url = self.plexauth.auth_url()
+        self.hass.helpers.event.async_call_later(
+            DELAY_CONFIGURE, self.hass.config_entries.flow.async_configure(self.flow_id)
```

Comment:
Help request: the inner call here causes pytest warnings about not being awaited, but with an await it hangs the flow. The intention is to continue flow execution 5 seconds after the external step starts and begin polling for a token.

A patch to set the delay to 0 and issuing `await hass.async_block_till_done()` in the tests still throws the warning.

Reply:
Why do we need to poll? How does the plex flow work? Is it oauth2?

Reply:
I originally had it written to use a callback URL passed to the Plex endpoint, but I wasn't able to reliably determine the proper base URL of every HA setup to guarantee it would always hit. I'm using this method instead as it was always reliable in my testing.

Reply:
Note that the current behavior works well and all CI tests pass, but there are warnings that I'd like to clean up.

Reply:
Did you test something like this?
```py
redirect_uri = f"{self.hass.config.api.base_url}{AUTH_CALLBACK_PATH}"
```

Reply:
Yes, but not everyone has confirmed their base URL correctly. There are also some edge cases where a NAT or proxy IP is needed to connect back to HA. If an unreachable destination is used in the callback URL the the flow fails to complete.

Reply:
Aren't those edge cases? We still support specifying a token directly in two ways.

Reply:
Yes, but a failure leads to a state that can't be detected (or continued without editing the URL in the external browser) and would be confusing for the user. The polling method mitigates those risks of a potential hard failure.

Reply:
@jjlawren did you put the `await hass.async_block_till_done()` inside the patch block ?

Reply:
Sorry, only read up now. I think that we should be able to use the base url. People need to make sure that it is configured correctly.

Reply:
Yes, I issued the blocking call inside the patch block. Warnings still persisted.

Since the Plex auth flow callback contains no auth information and is just a trigger to let us know a token can be requested, is there a benefit to use this over the polling method? It's slightly less efficient to poll, but using the callback exposes a non-zero chance of failing.

Reply:
@MartinHjelmare @balloob here's what the callback method would look like: [diff](https://github.com/jjlawren/home-assistant/compare/plex_external_auth...jjlawren:plex_external_auth_v2)

I think from the user's perspective the existing way in this PR has fewer edge cases that can go wrong, but I defer to you both on the preferred way forward.

Reply:
I suggest we use the callback way.

Reply:
Ok, switched to use a callback and updated tests accordingly. No more warnings 👍